File: C#/CreationalPatterns/CreationalPatterns.Builder/FirstHouse.cs
namespace CreationalPatterns.Builder
{
    internal class FirstHouse : Builder
    {
        public FirstHouse()
        {
            house = new House();
        }

        public override House BuildHouse()
        {
            house.BuildBase();
            house.BuildWalls();
            house.BuildFloors();
            house.BuildRoof();
            return house;
        }
    }
}

File: Java/creational_patterns/src/Builder/FirstHouse.java
package Builder;

public class FirstHouse extends Builder {
    public FirstHouse() {
        house = new House();
    }

    @Override
    public House builHouse() {
        house.buildBase();
        house.buildWalls();
        house.buildFloors();
        house.buildRoof();
        return house;
    }
}
